pydis_site/apps/api/viewsets/bot/user.py

```diff
@@ -118,4 +168,69 @@ class UserViewSet(BulkCreateModelMixin, ModelViewSet):
     """
 
     serializer_class = UserSerializer
-    queryset = User.objects
+    queryset = User.objects.all()
+    pagination_class = UserListPagination
+
+    def get_serializer(self, *args, **kwargs) -> ModelSerializer:
+        """Set Serializer many attribute to True if request body contains a list."""
+        if isinstance(kwargs.get('data', {}), list):
+            kwargs['many'] = True
+
+        return super().get_serializer(*args, **kwargs)
+
+    @action(detail=False, methods=["PATCH"], name='user-bulk-patch')
+    def bulk_patch(self, request: Request) -> Response:
+        """
+        Update multiple User objects in a single request.
+
+        ## Route
+        ### PATCH /bot/users/bulk_patch
+        Update all users with the IDs.
+        `id` field is mandatory, rest are optional.
+
+        #### Request body
+        >>> [
+        ...     {
+        ...     'id': int,
+        ...     'name': str,
+        ...     'discriminator': int,
+        ...     'roles': List[int],
+        ...     'in_guild': bool
+        ...     },
+        ...     {
+        ...     'id': int,
+        ...     'name': str,
+        ...     'discriminator': int,
+        ...     'roles': List[int],
+        ...     'in_guild': bool
+        ...     },
+        ... ]
+
+        #### Status codes
+        - 200: returned on success.
+        - 400: if the request body was invalid, see response body for details.
+        - 404: if the user with the given id does not exist.
+        """
+        queryset = self.get_queryset()
+        try:
+            object_ids = [item["id"] for item in request.data]
+        except KeyError:
+            # user ID not provided in request body.
+            resp = {
+                "Error": "User ID not provided."
```

Comment:
For consistency with the rest of DRF, it should be named `"detail"`.

```suggestion
                "detail": "The id field is missing from at least one object."
```

Reply:
Alternatively, you can raise a validation error with 

```py
raise ValidationError({"id": ["This field is required."]})
```

Which may or may not be more idiomatic here. It's in a list cause that's how DRF normally formats such errors. I am leaning more towards the `ValidationError` approach.